app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -993,6 +993,12 @@ class PluginApi {
    *
    */
   addUserMenuGlyph(glyph) {
+    if (this.getCurrentUser()?.redesigned_user_menu_enabled) {
```

Comment:
Thinking more broadly: does it make sense to gate deprecation warnings behind a user setting? We probably want to know about outdated use asap, especially when we'll have live deprecation metrics.

Reply:
True... it's a tricky problem. Right now plugins like discourse-assign implement both `registerUserMenuTab` and `addUserMenuGlyph`, so that it's functional on both old and new user menus. It's probably not fair to print a deprecation message given that `addUserMenuGlyph` is still required...

Maybe we skip the deprecation message for now, and just add a note in the comment above the method.

Reply:
Hm, good point. If it's required, then it's not deprecated 😅